src/main/java/cuchaz/enigma/utils/Utils.java

```diff
@@ -94,4 +97,14 @@ public static boolean getSystemPropertyAsBoolean(String property, boolean defVal
 		String value = System.getProperty(property);
 		return value == null ? defValue : Boolean.parseBoolean(value);
 	}
+
+	public static void delete(Path path) throws IOException {
+		if (path.toFile().exists()) {
+			Files.walk(path).sorted(Comparator.reverseOrder()).map(Path::toFile).forEach(File::delete);
```

Comment:
```java
Files.walk(path).sorted(Comparator.reverseOrder()).forEach(Files::delete);
```
Shorter.

If the sorting impacts the performance, you can use a file visitor to delete alternatively.

Reply:
Doesn't work, since Files::delete throws an IO exception.